src/RecommendedOptions.cmake

```diff
@@ -0,0 +1,136 @@
+# ENABLE_DEVELOPER_MODE: sets defaults appropriate for developers, this is defaulted to ON
+#  * WARNINGS_AS_ERRORS: ON
+#  * ENABLE_SANITIZER_ADDRESS: ON
+#  * ENABLE_CLANG_TIDY: ON for unix
+#  * ENABLE_SANITIZER_UNDEFINED: ON for unix
+#  * ENABLE_CPPCHECK: ON for unix
+
+# For non-developer builds
+#  -DENABLE_DEVELOPER_MODE:BOOL=OFF
+# Is recommended
+
+# In developer mode, all features have options that show up in the CMake GUI tools
+
+# recommended_project_options() macro enables all recommended defaults with appropriately
+# applied options from the GUI which are set
+
+# Any default can be overridden
+# set(<feature_name>_DEFAULT <value>) - set default for both user and developer modes
+# set(<feature_name>_DEVELOPER_DEFAULT <value>) - set default for developer mode
+# set(<feature_name>_USER_DEFAULT <value>) - set default for user mode
+
+option(ENABLE_DEVELOPER_MODE "Set up defaults for a developer of the project, and let developer change options" ON)
+
+if((CMAKE_CXX_COMPILER_ID MATCHES ".*Clang.*" OR CMAKE_CXX_COMPILER_ID MATCHES ".*GNU.*") AND NOT WIN32)
+  set(GCC_OR_CLANG ON)
+else()
+  set(GCC_OR_CLANG OFF)
+endif()
+
+if(CMAKE_GENERATOR MATCHES ".*Makefile*." OR CMAKE_GENERATOR MATCHES ".*Ninja*")
+  set(MAKEFILE_OR_NINJA ON)
+else()
+  set(MAKEFILE_OR_NINJA OFF)
+endif()
+
+include(CMakeDependentOption)
+
+# <option name>;<user mode default>;<developer mode default>;<description>
+set(options
+    "ENABLE_CACHE\;${MAKEFILE_OR_NINJA}\;${MAKEFILE_OR_NINJA}\;Enable ccache on Unix"
```

Comment:
Why should we disable cache on Windows?

Reply:
This doesn't disable it on windows, it requires Makefiles and Ninja, according to CMake's documentation:

![image](https://user-images.githubusercontent.com/234279/151567266-b6f8b7c2-b4c9-4af5-937f-7e3574a19a51.png)

Reply:
Updated code documentation to reflect why this default was chosen.